### ISSUE TITLE
Object constructor doesn't copy values from input object

### ISSUE THREAD

**Issue by Jezzamonn** [ruffle-rs/ruffle#2384]
Sort of out-of-spec, but in AS2 you can pass an Object into the Object constructor, and it'll cast it to an Object.

```
var obj = new Object({param: 0});
trace("obj.param = " + obj.param); // obj.param = 0
```

Redundant, but I suppose some games relied on this (the Stinkoman 20X6 game being one of them).

**In Ruffle, this prints `obj.param = undefined`.** Tested with the demo page.

I say out-of-spec because the AS2 manual doesn't mention this:
```
Object function
Object([value:Object]) : Object
Creates a new empty object or converts the specified number, string, or Boolean value to an
object. This command is equivalent to creating an object using the Object constructor (see
"Constructor for the Object class").
Availability: ActionScript 1.0; Flash Player 5
Parameters
value:Object [optional] - A number, string, or Boolean value.
Returns
Object - An object.
```

Also, I did some experimentation in the Flash Editor, and the object returned is not a copy of the parameters but the same instance. For example:

```
var firstObj = {param: 0};
var secondObj = new Object(firstObj);

trace("firstObj.param = " + firstObj.param); // firstObj.param = 0
trace("secondObj.param = " + secondObj.param); // secondObj.param = 0

firstObj.param = 1;
trace("firstObj.param = " + firstObj.param); // firstObj.param = 1
trace("secondObj.param = " + secondObj.param); // secondObj.param = 1
```

For Stinkoman 20X6 we can fix things on the game side by avoiding the redundant cast, because I'm able to update the game :)

Attached is a zip of the .swf file I used to test the ruffle behavior. Has the above code snippets.
[ruffle object constructor example.swf.zip](https://github.com/ruffle-rs/ruffle/files/5794630/ruffle.object.constructor.example.swf.zip)

**Comment by Herschel** [comment]
Same issue as #1635.

**Comment by Herschel** [comment]
Fixed by #2415.
